src/compiler/types.ts

```diff
@@ -1485,11 +1485,11 @@ module ts {
     // Class and interface types (TypeFlags.Class and TypeFlags.Interface)
     export interface InterfaceType extends ObjectType {
         typeParameters: TypeParameter[];           // Type parameters (undefined if non-generic)
-        declaredProperties: Symbol[];              // Declared members
-        declaredCallSignatures: Signature[];       // Declared call signatures
-        declaredConstructSignatures: Signature[];  // Declared construct signatures
-        declaredStringIndexType: Type;             // Declared string index type
-        declaredNumberIndexType: Type;             // Declared numeric index type
+        declaredProperties: Symbol[];              // Declared members (computed by resolveDeclaredMembers)
+        declaredCallSignatures: Signature[];       // Declared call signatures (computed by resolveDeclaredMembers)
+        declaredConstructSignatures: Signature[];  // Declared construct signatures (computed by resolveDeclaredMembers)
+        declaredStringIndexType: Type;             // Declared string index type (computed by resolveDeclaredMembers)
+        declaredNumberIndexType: Type;             // Declared numeric index type (computed by resolveDeclaredMembers)

```

Comment:
Yeah, I did that to begin with using the same pattern as `ResolvedType`, but it just felt overly complicated. My preference is the current approach, but I don't feel strongly about it.